### PR TITLE
Separate diagnostic edmf precomputed quants int 4 funcs

### PR DESCRIPTION
**Pull request by trontrytel**

### CODE REVIEW

File: src/cache/diagnostic_edmf_precomputed_quantities.jl [hunk 1]
```diff
@@ -632,21 +642,45 @@ function set_diagnostic_edmf_precomputed_quantities!(Y, p, t)
...
+        S_q_totʲ_level = F.field_values(F.level(ᶜS_q_totʲ, i_top))
+        S_e_totʲ_helper_level = F.field_values(F.level(ᶜS_e_totʲ_helper, i_top))
         @. S_q_totʲ_level = 0
         @. S_e_totʲ_helper_level = 0
     end
+    return nothing
+end
+
+"""
+    set_diagnostic_edmf_precomputed_quantities!(Y, p, t)
```
Comment (szy21) [inline src/cache/diagnostic_edmf_precomputed_quantities.jl:663]:
```suggestion
    set_diagnostic_edmf_precomputed_quantities_closures!(Y, p, t)
```

File: src/cache/diagnostic_edmf_precomputed_quantities.jl [hunk 2]
```diff
@@ -632,21 +642,45 @@ function set_diagnostic_edmf_precomputed_quantities!(Y, p, t)
...
+
+Updates the precomputed quantities stored in `p` for diagnostic edmfx.
```
Comment (szy21) [inline src/cache/diagnostic_edmf_precomputed_quantities.jl:665]:
```suggestion
Updates the environment closures in precomputed quantities stored in `p` for diagnostic edmfx.
```

File: src/cache/diagnostic_edmf_precomputed_quantities.jl [hunk 3]
```diff
@@ -617,10 +615,22 @@ function set_diagnostic_edmf_precomputed_quantities!(Y, p, t)
             turbconv_model,
         )
     end
+    return nothing
+end
+
+"""
+    set_diagnostic_edmf_precomputed_quantities_top_bc!(Y, p, t)
+
+Updates the precomputed quantities stored in `p` for diagnostic edmfx.
```
Comment (szy21) [inline src/cache/diagnostic_edmf_precomputed_quantities.jl:624]:
```suggestion
Updates the top boundary condition of precomputed quantities stored in `p` for diagnostic edmfx.
```

File: src/cache/diagnostic_edmf_precomputed_quantities.jl [hunk 4]
```diff
@@ -632,21 +642,45 @@ function set_diagnostic_edmf_precomputed_quantities!(Y, p, t)
...
     end
+    return nothing
+end
+
+"""
+    set_diagnostic_edmf_precomputed_quantities!(Y, p, t)
+
+Updates the precomputed quantities stored in `p` for diagnostic edmfx.
+"""
+function set_diagnostic_edmf_precomputed_quantities_closures!(Y, p, t)
```
Comment (szy21) [inline src/cache/diagnostic_edmf_precomputed_quantities.jl:667]: (Just want to note that it's more accurate to call it something like environment closures as the closures related to updraft, including entrainment, detrainment, nonhydrostatic pressure drag, and microphysics are in the integral. But I'm fine with the current name.)

**Review by szy21 — APPROVED** [review]
Thanks! I prefer to keep `Fields` to be consistent with the rest of the code base. But I'm fine with this change if @Sbozzolo thinks it's a good idea.

**Comment by Sbozzolo** [comment]
> Thanks! I prefer to keep `Fields` to be consistent with the rest of the code base. But I'm fine with this change if @Sbozzolo thinks it's a good idea.

`Fields` would make search-and-replace operations easier, but I can see that it is highly redundant, so I don't have strong opinions.

**Pull request merged**